added llvm 3.8.1 and 3.9.0 support

foerdi opened the pull request:
- added llvm 3.8.1 and 3.9.0 support
  based on changes of the changelogs:
    - http://llvm.org/releases/3.8.1/docs/ReleaseNotes.html
    - http://llvm.org/releases/3.9.0/docs/ReleaseNotes.html
- removed duplicated license entry in dub.json

Calrama commented:
Hey, thank you for the work, have you checked with respective C header diffs that your changes are complete?
Otherwise, two things I'd like before I can merge this:
1: Please move the changes to ```samples/fibonacci.d``` into a different commit with an appropriate message (probably a bugix?), as it seems to me that they don't relate to the 3.8.1/3.9.0 update.
2: While I like the idea with `removedIn***` [https://github.com/Calrama/llvm-d/pull/12/commits/087476a13756ab2f7f1b27e5f1fd5fb5749b2f68#diff-1cf89943c9347f1fb3ebbbafaf1bd549R225], I prefer homogeneous code more: Please either remove these enums and use the values directly or add `addedIn` / `removedIn` for all the other changes.

foerdi commented:
in the last week I was busy, but now:

- fixed commit tree
- fixed code style
- added missing changes in (llvm 3.9.0)

let me know if something is wrong with the PR :)

Calrama commented:
LGTM, thank you for the contribution.
